src/Microsoft.Health.Fhir.Core/Extensions/RawResourceExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using EnsureThat;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Serialization;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Models;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Core.Extensions
+{
+    public static class RawResourceExtensions
+    {
+        /// <summary>
+        /// Converts the RawResource to an ITypedElement
+        /// </summary>
+        /// <param name="rawResource">The RawResource to convert</param>
+        /// <param name="modelInfoProvider">The IModelInfoProvider to use when converting the RawResource</param>
+        /// <returns>An ITypedElement of the RawResource</returns>
+        public static ITypedElement ToITypedElement(this RawResource rawResource, IModelInfoProvider modelInfoProvider)
```

Comment:
By switching this around it could become an overload, that might make discovery easier, thoughts?

```suggestion
        public static ITypedElement ToTypedElement(this IModelInfoProvider modelInfoProvider, RawResource rawResource)
```

Reply:
I agree, I have updated it as an overload on ModelInfoProvider.